sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidSqlValidator.java

```diff
@@ -432,11 +456,18 @@ private RelDataType validateTargetType(
         fields.add(Pair.of(colName, sourceField.getType()));
         continue;
       }
-      RelDataType relType = typeFactory.createSqlType(SqlTypeName.get(sqlTypeName));
-      fields.add(Pair.of(
-          colName,
-          typeFactory.createTypeWithNullability(relType, true)
-      ));
+      if (NullHandling.replaceWithDefault()) {
```

Comment:
There is a test that includes a String column, and it seems to produce a targetType of VARCHAR NOT NULL when this is false, and a nullable VARCHAR when true, so I think this is handled correctly, but let me know if you think otherwise.

Reply:
for string types, default value mode treats `null` and `''` interchangeably, so it probably should be marked as nullable? tbh its not super well defined, but we do always mark string values as nullable in default value mode since the empty strings behave as null.

For numbers though default value mode effectively means that they don't exist.

This default value mode is deprecated now, so we probably don't have to spend too much time thinking about how things should behave, so we should probably match the computed schema stuff (strings always nullable in either mode, numbers only nullable in sql compatible mode)

Reply:
Thanks! Fixed

Reply:
I think if that's the case the logic should not live *here* in this file - instead it should be in a more central location.... its bad to see these conditions flow everywhere